### PR TITLE
initial commit to query cases

### PR DESCRIPTION
**Pull request by breid1313**
tested on my end and working at this point.

do note that at this time, if the user wants to query multiple incident fields, the conditions are joined with an AND operand. for example, if the conditions supplied are `[ ["name", "splunk", "contains"], ["description", "test", "contains"] ]` the result returned will be incidents that satisfy both conditions. we should be supporting OR in the future

### CODE REVIEW

File: plugins/modules/cp4s/cp4s_query_incidents.py [hunk 1]
```diff
@@ -0,0 +1,262 @@
+
+# Copyright: (c) 2021, Ryan Gordon
```
Comment (Ryan-Gordon1) [inline plugins/modules/cp4s/cp4s_query_incidents.py]: Add your name to the copyright for this file. ;)

File: plugins/modules/cp4s/cp4s_query_incidents.py [hunk 2]
```diff
@@ -0,0 +1,262 @@
...
+
+# Query open incidents by name, exact match
+- name: Test query on CP4S cases
+  ryan_gordon1.cloud_pak_for_security.cp4s_query_incidents:
+    conditions: ["name", "example_name", "equals"]
+
+# Query closed incidents by name
+- name: Test query on CP4S cases
+  ryan_gordon1.cloud_pak_for_security.cp4s_query_incidents:
+    conditions: ["name", "example_name", "equals"]
```
Comment (Ryan-Gordon1) [inline plugins/modules/cp4s/cp4s_query_incidents.py]: is this like a [key, value, operator] syntax?

**Review by Ryan-Gordon1 — COMMENTED** [review]
Left two comments. Great work!

File: plugins/modules/cp4s/cp4s_query_incidents.py (hunk 1 above)
Comment (breid1313) [reply plugins/modules/cp4s/cp4s_query_incidents.py]: thanks :)

File: plugins/modules/cp4s/cp4s_query_incidents.py (hunk 2 above)
Comment (breid1313) [reply plugins/modules/cp4s/cp4s_query_incidents.py]: basically. to query cases in resilient on a condition, you need to specify an incident field ie name or description, a value, and a method ie equals for an exact match.

I updated some of the doc terminology, so this should be a bit more clear. I've also directed the user to the resilient/cases rest API docs, as they will need that and/or previously development experience with the product to know whats available to them

**Review by Ryan-Gordon1 — APPROVED** [review]
LGTM. Great Job! Merging

**Pull request merged**